tests/unit/test_data_subset.py

```diff
@@ -40,3 +42,43 @@ def test_box_subset():
     # Check we have the points we asked for
     assert iris.util.array_equal(xcoord.points, [0, 2000])
     assert iris.util.array_equal(ycoord.points, [-2000, 0, 2000, 4000])
+
+def test_polygon_subset():
+    # Define a test polygon (an extremely simple representation of Exeter)
+    shape = shapely.geometry.Polygon([
+        (289271.9, 93197.0),
+        (289351.3, 95110.1),
+        (293405.1, 96855.0),
+        (296721.1, 94960.3),
+        (297165.1, 86966.9),
+        (294181.6, 89357.2),
+        (291388.0, 89272.6),
+    ])
+
+    # Define corresponding mask (note: this "looks" upside down compared
+    # to how it would be plotted)
+    expected_mask = np.array(
+        [[1, 1, 1, 1, 0],
+         [1, 1, 0, 0, 0],
+         [0, 0, 0, 0, 0],
+         [0, 0, 0, 0, 1],
+         [0, 0, 0, 0, 1],
+         [0, 0, 0, 0, 1]]
+    )
+
+    # Create example dataset
+    ds = DataSubset(
+        None,
+        "aqum",
+        os.path.join(SAMPLEDIR, "model", "aqum_hourly_o3_20200520.nc"),
+        shape=shape,
+    )
+    cube = ds.as_cube()
+
+    # Check we have the right mask (on a 2d slice of this 3d cube)
+    subcube = next(cube.slices_over("time"))
+    assert iris.util.array_equal(subcube.data.mask, expected_mask)
+
+    # Simple data check, which, as the mask is taken into account, should
+    # be a pretty reliable test
+    assert round(cube.data.mean(), 8) == 57.66388811
```

Comment:
Can you turn these into two separate tests please? This looks like an appropriate case for a test class with a setup_class bit and then these two tests underneath.  It just makes it a bit easier to find bugs when the tests are broken down more and have super-descriptive titles, like 'test_polygon_subset_mask' and 'test_polygon_subset_data'.

Reply:
Done this now. Must admit I'm a little sceptical about using classes to group tests, but there's no real alternative, that's just how it is.